backends/arm/test/passes/test_tag_io_quant_pass.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2024 Arm Limited and/or its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import logging
+import unittest
+
+import torch
+from executorch.backends.arm.arm_backend import generate_ethosu_compile_spec
+from executorch.backends.arm.test import common
+
+from executorch.backends.arm.test.tester.arm_tester import ArmTester
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class Linear(torch.nn.Module):
+    def __init__(
+        self,
+        in_features: int,
+        out_features: int = 3,
+        bias: bool = True,
+    ):
+        super().__init__()
+        self.inputs = (torch.ones(1, 1, 1, in_features),)
+        self.fc = torch.nn.Linear(
+            in_features=in_features,
+            out_features=out_features,
+            bias=bias,
+        )
+
+    def get_inputs(self):
+        return self.inputs
+
+    def forward(self, x):
+        return self.fc(x)
+
+
+class TestTagIOQuantPass(unittest.TestCase):
+
+    def _tosa_BI_u55_pipeline(self, module: torch.nn.Module):
+        (
+            ArmTester(
+                module,
+                inputs=module.get_inputs(),
+                compile_spec=generate_ethosu_compile_spec(
+                    "ethos-u55-128",
+                    permute_memory_to_nhwc=True,
+                    quantize_io=True,
```

Comment:
This is one way to do it. But why hide it from the user? I would just run them at the top level i.e.

```
...
e = to_edge()
e = helper_run_pass(e, pass_foo) # new method
e = partition(my_delegate_partitioner())
e = helper_run_pass(e, pass_bar)
...
```

In theory delegate specific passes should only run on the "partitioned" subgraph i.e. inside the `preprocess()` to make sure the graph you are mutating has no external side effect.

This is different, and I understand the rationale, and this is mutating the graph *before* the partitioning. I am not sure if we want to hide this mutation from the user. 

What do you think?

Reply:
> But why hide it from the user? 


The reasoning is that different compile_specs may require different passes and applying all the passes at the top level would leave all the responsibility on the end-user. With this pass manager, all the user needs to do is to configure the compile_spec and then the passes would be applied for "free".

On the second part I think it is fine that this specific pass (TagIOQuantPass) is hidden since it does not change the graph, it only makes sure that the delegate does not consume certain nodes. Pre-partition passes that does mutate the graph should probably not be hidden though as you say...

Reply:
well, you can still do it outside, I am not sure which one is better

```
e = helper_run_pass(e, pass, compile_spec)
```
@mergennachin  - what do you think?